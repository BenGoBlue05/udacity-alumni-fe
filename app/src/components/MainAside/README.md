## MainAside Component
A component that ...

### Example

```js
<MainAside />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
