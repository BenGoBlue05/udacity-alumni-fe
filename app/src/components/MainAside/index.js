import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Heading from 'grommet-udacity/components/Heading';
import Image from 'grommet-udacity/components/Image';
import Box from 'grommet-udacity/components/Box';
import Button from 'grommet-udacity/components/Button';
import Menu from 'grommet-udacity/components/Menu';
import EditIcon from 'grommet-udacity/components/icons/base/Edit';
import ArticleIcon from 'grommet-udacity/components/icons/base/Article';
import ImageIcon from 'grommet-udacity/components/icons/base/Image';

const MainAside = ({
  user,
}) => (
  <Box
    basis="1/3"
    pad={{ vertical: 'large', horizontal: 'small' }}
    align="center"
    className={styles.aside}
  >
   {user &&
     <div>
       <div className={styles.avatarWrapper}>
         <Image
           className={styles.avatarImage}
           size="medium"
           src={user.avatar}
         />
       </div>
       <Box
         className={styles.careerResourcesBlurb}
         align="center"
         pad={{ horizontal: 'small', vertical: 'small' }}
       >
         <Heading tag="h3" align="center">
           {`Hello, ${user.name}!`}
         </Heading>
       </Box>
       <Box
         className={styles.careerResourcesBlurb}
         basis="2/3"
         align="center"
         pad={{ horizontal: 'small', vertical: 'small' }}
       >
         <Button
           label="Post an Article"
           onClick={e => e}
           plain
           href="/admin/cms?action=new"
           icon={<EditIcon />}
         />
         <Button
           label="Manage Articles"
           onClick={e => e}
           plain
           href="/admin/content-dashboard"
           icon={<ArticleIcon />}
         />
       </Box>
       <Box className={styles.widgetBox}>
         <Heading align="center" tag="h3">
           Widgets
         </Heading>
         <Box className={styles.innerWidgetBox}>
           <Menu inline>
             <Button
               label="Carousel"
               onClick={e => e}
               plain
               href="/admin/carousel"
               icon={<ImageIcon />}
             />
           </Menu>
         </Box>
       </Box>
     </div>
   }
  </Box>
);

MainAside.propTypes = {
  user: PropTypes.object.isRequired,
};

export default cssModules(MainAside, styles);
