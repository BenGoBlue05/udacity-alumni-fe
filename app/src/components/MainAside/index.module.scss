.avatarWrapper {
  height: 300px;
  width: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #f5f5f5;
}

.avatarImage {
  width: 200px;
  height: auto;
  max-height: 300px;
}

.aside {
  margin-bottom: 14px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(46,61,73,.2);
  border: 1px solid #dbe2e8;
  margin-left: 30px;
  @media screen and (max-width: 1260px) {
    display: none !important;
  }
}

.innerWidgetBox {
  border: 1px solid #dbe2e8;
}
