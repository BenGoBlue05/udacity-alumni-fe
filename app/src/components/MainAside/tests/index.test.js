import MainAside from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<MainAside />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <MainAside
        avatarImage="http://helloworld.com"
        careerResourcesParagraph="lorem ipsum"
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
