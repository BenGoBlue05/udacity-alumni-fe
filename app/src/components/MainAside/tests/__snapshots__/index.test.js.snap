exports[`<MainAside /> should render with default props 1`] = `
<Box
  align="center"
  announce={false}
  basis="1/4"
  direction="column"
  focusable={true}
  pad={
    Object {
      "horizontal": "small",
      "vertical": "large"
    }
  }
  responsive={true}
  tag="div">
  <div>
    <div>
      <Image
        size="medium"
        src="http://github.com/image.png" />
    </div>
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      pad={
        Object {
          "horizontal": "small",
          "vertical": "small"
        }
      }
      responsive={true}
      tag="div">
      <Heading
        align="center"
        tag="h3">
        Hello, David Harris!
      </Heading>
    </Box>
    <Box
      align="center"
      announce={false}
      basis="2/3"
      direction="column"
      focusable={true}
      pad={
        Object {
          "horizontal": "small",
          "vertical": "small"
        }
      }
      responsive={true}
      tag="div">
      <Button
        href="/admin/cms?action=new"
        icon={
          <Edit
            a11yTitleId="edit-title"
            responsive={true} />
        }
        label="Post an Article"
        onClick={[Function onClick]}
        plain={true}
        type="button" />
      <Button
        href="/admin/content-dashboard"
        icon={
          <Article
            a11yTitleId="article-title"
            responsive={true} />
        }
        label="Manage Articles"
        onClick={[Function onClick]}
        plain={true}
        type="button" />
    </Box>
    <Box
      announce={false}
      direction="column"
      focusable={true}
      pad="none"
      responsive={true}
      tag="div">
      <Heading
        align="center"
        tag="h3">
        Widgets
      </Heading>
      <Box
        announce={false}
        direction="column"
        focusable={true}
        pad="none"
        responsive={true}
        tag="div">
        <Menu
          closeOnClick={true}
          direction="column"
          dropAlign={
            Object {
              "left": "left",
              "top": "top"
            }
          }
          inline={true}
          pad="none">
          <Button
            href="/admin/carousel"
            icon={
              <Image
                a11yTitleId="image-title"
                responsive={true} />
            }
            label="Carousel"
            onClick={[Function onClick]}
            plain={true}
            type="button" />
        </Menu>
      </Box>
    </Box>
  </div>
</Box>
`;
